### ISSUE TITLE
chore(web): resolve missing outputs in web build

### ISSUE THREAD

**Issue by mcdurdin** [keymanapp/keyman#8358]
Doing a full `/web/build.sh` I get the following warnings:

```
## [web/src/tools/building/sourcemap-root] build was successful, but output does not exist
## [web/src/tools/building/sourcemap-root] Expected output: 'web/build/tools/sourcemap-root/index.js'.

## [web/src/engine] build:device-detect was successful, but output does not exist
## [web/src/engine] Expected output: 'web/src/engine/build/engine/device-detect/obj/index.js'.

## [web/src/engine] build:element-wrappers was successful, but output does not exist
## [web/src/engine] Expected output: 'web/src/engine/build/engine/element-wrappers/obj/index.js'.

## [web/src/engine] build:main was successful, but output does not exist
## [web/src/engine] Expected output: 'web/src/engine/build/engine/main/obj/keymanweb.js'.

## [web/src/tools/testing/recorder] build was successful, but output does not exist
## [web/src/tools/testing/recorder] Expected output: 'web/build/tools/testing/recorder/index.js'.

## [web/src/tools/building/sourcemap-root] build was successful, but output does not exist
## [web/src/tools/building/sourcemap-root] Expected output: 'web/build/tools/sourcemap-root/index.js'.
```

Note how `sourcemap-root` is flagged twice!

**Comment by jahorton** [comment]
Yeah... this is why I made the warning; it's something since gotten wrong a lot.

The ES-module feature branch, at least, is solid in this regard.

... this might actually be related to a noted (private-channel) issue on `master` with Developer builds of Web - the weird sourcemap paths thing we were seeing can definitely be caused by the following scenario:

- C depends on A, B
- B depends on A
- B is built first, which triggers a build of A
- A is rebuilt
- C is built without rebuilding B

The two different timestamps for A lead the TypeScript compiler to create a forcible distinction between the two versions, which results in the odd path in the sourcemaps that was causing Closure to fall over.

**Comment by mcdurdin** [comment]
The Closure thing may be unrelated -- see https://github.com/keymanapp/keyman/pull/8394, because Developer was requesting `--no-minify` but was _still_ getting partial minification due to a typo in web/build.sh.

> * A is rebuilt
> * C is built without rebuilding B

TypeScript is supposed to be able to figure these deps out in `-b` mode. If Closure is overwriting the TypeScript outputs, then we should try and resolve that by ensuring that it always writes to a different output file, and then tsc won't get confused.

**Comment by jahorton** [comment]
> If Closure is overwriting the TypeScript outputs, then we should try and resolve that by ensuring that it always writes to a different output file, and then tsc won't get confused.

I'll go ahead and state that this part is not happening.  What I'm referring to, and had seen, was a mangling of sourcemapped `source` files with paths that `tsc` mangled.  _That_ mangle then cascaded into an actual error thrown by Closure during minification runs... which, granted, Developer builds can skip via that flag.

Sounds like yet another case of two separate underlying errors being required for a visible error to surface.  Fixing one eliminates the symptoms, but leaving the other intact could leave a landmine for later.  Except... the non-`no-minify` part will be addressed as part of `feature-esmodule-web-engine`.  So... I guess it is fine to not address the `tsc` mangling on `master` should we wish to ignore it / ignore this issue.

**Comment by mcdurdin** [comment]
> was a mangling of sourcemapped `source` files with paths that `tsc` mangled.

OK, I'm not really understanding what exactly is happening here -- does this look like a bug in `tsc` or something else?

Given you say that changes with esmodules should resolve this, happy to just proceed without further changes.

**Comment by jahorton** [comment]
A fix is already included within `feature-esmodule-web-engine`, so this will be fully fixed once #8560 lands.

Fixes likely happened on a per-module basis during some of the earlier 🧩 PRs, and then re-fixed when resolving some of the larger merge conflicts when big updates occurred with master.
